### PR TITLE
Nav mesh for each scene

### PR DESCRIPTION
**Pull request by JorxSS**
WHAT'S HOT?

* Now we can have multiple navmeshes one from each scene

### CODE REVIEW

File: Engine/AI/NavMesh.cpp [hunk 1]
```diff
@@ -764,8 +770,58 @@ void NavMesh::SaveNavMesh(unsigned char* nav_data, unsigned int nav_data_size) c
...
+	}
+
+
+	is_mesh_computed = true;
+	duDebugDrawNavMeshWithClosedList(&m_dd, *nav_mesh, *nav_query, nav_mesh_draw_flags);
+	m_dd.GenerateBuffers();
+
+}
+
+void NavMesh::LoadNavMesh(const std::string nav_mesh_name)
```
Comment (Dartemiss) [inline Engine/AI/NavMesh.cpp:819]: Referece my pana

File: Engine/EditorUI/Panel/PanelNavMesh.h [hunk 2]
```diff
@@ -4,6 +4,7 @@
 #include "Panel.h"
 #include <MathGeoLib/MathGeoLib.h>
 #include <string>
+#include "AI/NavMesh.h"
```
Comment (Dartemiss) [inline Engine/EditorUI/Panel/PanelNavMesh.h:7]: CS

File: Engine/Module/ModuleAI.cpp [hunk 3]
```diff
@@ -39,3 +39,17 @@ ENGINE_API bool ModuleAI::FindNextPolyByDirection(float3& position, float3& next
 {
 	return nav_mesh.FindNextPolyByDirection(position, next_position);
 }
+
+void ModuleAI::SetNavMesh(const std::string nav_mesh_file)
```
Comment (Dartemiss) [inline Engine/Module/ModuleAI.cpp:43]: Reference my friend

**Review by Dartemiss — CHANGES_REQUESTED** [review]
GJ!
But it doesn't work (crashes) when trying to bake a navmesh from a scene that is not saved or new.

**Review by JoanStinson — APPROVED** [review]
1483897430_vR8qCsg png

**Comment by JorxSS** [comment]
It will take more time so, I will re open this in the future

**Pull request closed without merging**